upb/def.c

```diff
@@ -721,6 +722,35 @@ const char *upb_fielddef_name(const upb_fielddef *f) {
   return upb_def_fullname(upb_fielddef_upcast(f));
 }
 
+bool upb_fielddef_getjsonname(const upb_fielddef *f, char *buf) {
```

Comment:
why is this a method on fielddef instead of a different method taking any const char* name and camel casing?

If it has to be a method on fielddef, perhaps separate the transformation to a separate helper method that might be useful elsewhere for json